File: leetCode/python/p9_PalindromeNumber/lc9.py
import time

class Solution:
    """
    Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward.
    """
    def isPalindrome(self, x: int) -> bool:
        if (x != 0 and x % 10 == 0) or x < 0:
            return False
        
        if self.flip_number(x) == x:
            return True
        else:
            return False 

    def flip_number(self, x: int) -> int:  
        sol = 0 
    
        while x != 0:
            sol = sol * 10 + (x % 10)
            x = x // 10 

        return sol

    def test_function(self, x: int, sol: bool, test_num: int): 
        start_time = time.time() 
        tmp = self.isPalindrome(x)
        end_time = time.time()
        
        if tmp == sol:
            print('\nTEST' + str(test_num) + ': PASSED IN ' + '{:.2f}us'.format((end_time - start_time) * 1000000))            
        else:
            print('\nTEST FAILED') 

s = Solution()

# Test

s.test_function(121, True, 1)
s.test_function(-121, False, 2)
s.test_function(10, False, 3)
s.test_function(-101, False, 4)

File: leetCode/c/p9_PalindromeNumber/lc9.c
/* Determine whether an integer is a palindrome. An integer is a palindrome when it reads the same backward as forward */

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <time.h>

int flipNumber(int x) {
   
    int sol = 0;
 
    while(x != 0) {
        sol = sol * 10 + (x % 10);
        x /= 10;
    }
    
    return sol;  
}

bool isPalindrome(int x){
    
    if((x != 0 && x % 10 == 0) || x < 0) 
        return false;

    if(x == flipNumber(x)) 
        return true;
    else 
        return false;
}

/* Test */

void testFunction(int x, bool sol, int testNum) {
    
    bool tmp;
    double timeSpent;

    clock_t tic = clock();
 
    tmp = isPalindrome(x);
  
    clock_t toc = clock(); 
    
    timeSpent = (double) (toc - tic) * 1000000 / CLOCKS_PER_SEC;
 
    printf("\nTEST%d: ", testNum);
 
    if(tmp == sol) 
        printf("PASSED IN %.2fus\n", timeSpent);
    else  
        printf("FAILED\n");
}
 
int main(void) {
    
    testFunction(121, true, 1);
    testFunction(-121, false, 2);
    testFunction(10, false, 3);
    testFunction(-101, false, 4);

    return EXIT_SUCCESS;
}
